doc/developers-guide/Hooks-and-handlers.md

```diff
@@ -70,16 +66,9 @@ If you haven't encountered the term _fold_ before, think of it as _reduce_ (like
 
 ### Using accumulators
 
-Frankly speaking, we already mislead you twice.
-First, usage of `ejabberd_hooks:run` is strongly discouraged and some day will not be supported.
-Second, you should not use a list as an accumulator.
-
-MongooseIM is undergoing a major architectural change: we are switching to `run_fold` everywhere, and using a dedicated data structure to
-accumulate results(see ["Accumulators"](accumulators.md)).
-This data structure is implemented in `mongoose_acc` module, which has a map-like interface: it
-has `get/2`, `get/3`, `put/3` etc.
-It is instantiated with an incoming stanza, passed along throughout the processing
-chain, supplied to and returned from hook calls, and terminated when stanza is leaving MongooseIM.
+MongooseIM uses a dedicated data structure to accumulate results(see ["Accumulators"](accumulators.md)).
+This data structure is implemented in `mongoose_acc` module, which has a map-like interface: it has `get/2`, `get/3`, `put/3` etc.
```

Comment:
This data structure is implemented in the `mongoose_acc` module, which has a map-like interface: it has `get/2`, `get/3`, `put/3` etc.